sqlserver/datadog_checks/sqlserver/connection_errors.py

```diff
@@ -0,0 +1,143 @@
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import re
+from enum import Enum
+
+try:
+    import pyodbc
+except ImportError:
+    pyodbc = None
+
+try:
+    import adodbapi
+    from adodbapi.apibase import OperationalError
+    from pywintypes import com_error
+except ImportError:
+    adodbapi = None
+
+
+class SQLConnectionError(Exception):
+    """Exception raised for SQL instance connection issues"""
+
+    pass
+
+
+class ConnectionErrWarning(Enum):
```

Comment:
Since we're not emitting warnings anymore, it might make sense to rename this class to `ConnectionError` or `ConnectionErrorCode`?